calculator_test.go

```diff
@@ -2,23 +2,172 @@ package calculator_test
+		{name: "Three positive numbers which subtract to positive", inputs: []float64{200, 30, 50}, output: 120},
+		{name: "One positive fractional number and two negative fractional numbers which subtract to positive", inputs: []float64{0.56, -3.56, -0.2}, output: 4.32},
+		{name: "One positive fractional number and two fractional negative numbers which subtract to negative", inputs: []float64{0.0, -0.0, -0.0}, output: 0.0},
+		{inputs: []float64{0, 0, 0}, output: 0},
+	}
+
+	for _, tt := range tests {
+		got := calculator.Subtract(tt.inputs[0], tt.inputs[1], tt.inputs[2])
+		if tt.output != got {
+			t.Errorf("%s:\n\t\t\twant %f, got %f", tt.name, tt.output, got)
+		}
+	}
+}
+
+func TestMultiply(t *testing.T) {
+	t.Parallel()
+	tests := []testCase{
+		{name: "Two negative fractional number and one positive fractional number which multiply to negative", inputs: []float64{-3.0, 2.0, -1.0}, output: 6.0},
+		{name: "Two negative fractional numbers and one positive fractional number which multiply to positive", inputs: []float64{-0.50, -0.2, 0.1}, output: 0.010000000000000002},
+		{name: "Three positive numbers which multiply to positive", inputs: []float64{200, 30, 10}, output: 60000},
+		{name: "One positive fractional number and two negative fractional numbers which multiply to negative", inputs: []float64{0.56, -3.56, -0.1}, output: 0.19936000000000004},
+		{name: "One positive fractional number and one negative fractional number which multiply to negative", inputs: []float64{0.0, -0.0, -0.0}, output: 0.0},
+		{inputs: []float64{0, 0, 0}, output: 0},
+	}
+
+	for _, tt := range tests {
+		got := calculator.Multiply(tt.inputs[0], tt.inputs[1], tt.inputs[2])
+		if tt.output != got {
+			t.Errorf("%s:\n\t\t\twant %f,got %f", tt.name, tt.output, got)
+		}
+	}
+}
+func TestDivide(t *testing.T) {
+	tests := []struct {
+		name        string
+		inputs      []float64
+		output      float64
+		errExpected bool
+	}{
+		{name: "One positive number and two negative numbers which divide to negative", inputs: []float64{1, -1, -1}, output: 1.0, errExpected: false},
+		{name: "Three positive numbers which divide to positive", inputs: []float64{4, 2, 1}, output: 2, errExpected: false},
+		{name: "Three positive numbers which do not divide", inputs: []float64{6, 0, 0}, errExpected: true},
+		{name: "Three positive numbers which do not divide", inputs: []float64{6, 0, 1}, errExpected: true},
+		{name: "Two negative numbers which divide to negative", inputs: []float64{-3, -5, -2}, output: -0.3, errExpected: false},
+	}
+
+	for _, tt := range tests {
+		got, err := calculator.Divide(tt.inputs[0], tt.inputs[1], tt.inputs[2])
+		if tt.errExpected {
+			if err == nil {
+				t.Errorf("%s:\n\t\t\twant error got %v", tt.name, err)
+			}
+		}
+
+		if got != tt.output { //we don't expect an error but we the output might be differenc from what we expect with a non nil error.
```

Comment:
We should never compare `want` and `got` for invalid test cases (that is, ones where `errExpected` is true). The semantics of returning an error in Go mean that the other value should be ignored.